### PR TITLE
dev-qt/qtwebengine: add ~ppc64 keyword and patchset

### PR DESCRIPTION
**Pull request by gyakovlev**
Hi =)

It's me as promised.
Here's how I think it can be handled more or less safely. I'll maintain patchset, it's very unlikely to change. I used it for entire lifetime of qt-5.13 and 5.14 and not a single change was needed.

No extra keywords planned this time to give it some time to rest in the tree, and see how things go.
maybe just Falkon and unmask webengine useflag for kdecore-meta only, so people can get a working browser.

the only browser that works without patching on ppc64 currently is firefox, but it lacks JS Jit, so JS heavy sites can be laggy. Having falkon as an options is a nice bonus.

I also maintain chromium patchset for ppc64le, which is on version 83 so far, so I have some time discover issues with patches, since qtwebengine is always behind.

Package-Manager: Portage-2.3.100, Repoman-2.3.22
Signed-off-by: Georgy Yakovlev <gyakovlev@gentoo.org>

### CODE REVIEW

**Comment by gyakovlev** [comment]
actually, patchset is a bit dirty, need to get rid of .orig files, so don't merge, let's just discuss =)

File: dev-qt/qtwebengine/qtwebengine-5.15.0.ebuild [hunk 1]
```diff
@@ -9,9 +9,13 @@ inherit multiprocessing python-any-r1 qt5-build
 DESCRIPTION="Library for rendering dynamic web content in Qt5 C++ and QML applications"
 
 if [[ ${QT5_BUILD_TYPE} == release ]]; then
-	KEYWORDS="~amd64 ~arm ~arm64 ~x86"
+	KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 ~x86"
 fi
 
+# patchset unlikely to change on minor bumps, hardcode version
+# ppc64 patchset maintained by gyakovlev@gentoo.org
```
Comment (Pesa) [inline dev-qt/qtwebengine/qtwebengine-5.15.0.ebuild:16]: do we have a way to express this in `metadata.xml`?

File: dev-qt/qtwebengine/qtwebengine-5.15.0.ebuild [hunk 2]
```diff
@@ -9,9 +9,13 @@ inherit multiprocessing python-any-r1 qt5-build
...
+SRC_URI+=" ppc64? ( https://dev.gentoo.org/~gyakovlev/distfiles/${PN}-5.15.0-ppc64.tar.xz )"
```
Comment (Pesa) [inline dev-qt/qtwebengine/qtwebengine-5.15.0.ebuild:17]: we usually declare `SRC_URI` before `KEYWORDS`

File: dev-qt/qtwebengine/qtwebengine-5.15.0.ebuild [hunk 3]
```diff
@@ -85,6 +89,11 @@ PATCHES=(
 )
 
 src_prepare() {
+	if use ppc64; then
+		# ppc64 patches may lag behind for couple of days, do not block other arches
+		# ping gyakovlev@gentoo.org if you have any problems on ppc64
```
Comment (Pesa) [inline dev-qt/qtwebengine/qtwebengine-5.15.0.ebuild:94]: same here, can this be in `metadata.xml`?

File: dev-qt/qtwebengine/qtwebengine-5.15.0.ebuild [hunk 4]
```diff
@@ -85,6 +89,11 @@ PATCHES=(
...
+		eapply "${WORKDIR}/${PF}-ppc64"
```
Comment (Pesa) [inline dev-qt/qtwebengine/qtwebengine-5.15.0.ebuild:95]: If you expect the patchset to change rarely, using `PF` here doesn't seem right

File: dev-qt/qtwebengine/qtwebengine-5.15.0.ebuild (hunk 1 above)
Comment (gyakovlev) [reply dev-qt/qtwebengine/qtwebengine-5.15.0.ebuild:16]: sure, I can add myself there.

File: dev-qt/qtwebengine/qtwebengine-5.15.0.ebuild (hunk 4 above)
Comment (gyakovlev) [reply dev-qt/qtwebengine/qtwebengine-5.15.0.ebuild:95]: true, I have version in  the filenames and will just name the directory inside tarball as `qtwebengine-ppc64`

**Comment by gyakovlev** [comment]
> I'm ok with accepting this patchset in principle, given that you can be considered the "upstream" for ppc64 support in chromium. But see the ebuild comments.
> Do you maintain a branch or fork where you develop/rebase these patches? Might be useful to mention it somewhere.
patches are from here

 https://github.com/chromium-ppc64le/ 

and that's the upstream for ppc64 support, it's also used by ungoogled-chromium, electron port and others.
Everything is being slowly slowly upstreamed, a lot of work in progress behind the scenes.

I maintain local patches for myself and some users. No git right now, but there will be a repo soon, it's work in progress.

Since I could not get ppc64 patches and keywords get to gentoo (chromium moves way to fast, patches lag a bit), current gentoo chromium ebuild only supports ppc64 glue, it has some logic in it, but does not have keywords. Users run a script that downloads correct patches to build on gentoo and place them to `/etc/portage/patches/..` to build it.

File: dev-qt/qtwebengine/qtwebengine-5.15.0.ebuild [hunk 5]
```diff
@@ -85,6 +88,9 @@ PATCHES=(
 )
 
 src_prepare() {
+	if use ppc64; then
+		eapply "${WORKDIR}/${PN}-ppc64"
+	fi
```
Comment (Pesa) [inline dev-qt/qtwebengine/qtwebengine-5.15.0.ebuild:93]: (nit) add an empty line after this

Comment (gyakovlev) [reply dev-qt/qtwebengine/qtwebengine-5.15.0.ebuild:93]: sure, force-pushed.

**Pull request closed without merging**